buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -1442,7 +1447,7 @@ class BeamModulePlugin implements Plugin<Project> {
       JavaPerformanceTestConfiguration configuration = it ? it as JavaPerformanceTestConfiguration : new JavaPerformanceTestConfiguration()
 
       // Task for running integration tests
-      project.task('integrationTest', type: Test) {
+      def itTask = project.task('integrationTest', type: Test) {
```

Comment:
I'm not very familiar with this file, but it seems like we're now configuring runnerv2 tasks (e.g. cleanup tasks, building container images) based on various settings (e.g. pipelineOptions, the gradle 'experiment' flag) here and also in various build.gradle files.

Should we be picking one over the other? For example, the PR you're reviewing of mine that contains IT tests for examples, configures the tests in the build.gradle file. It seems strikingly similar to what is going on here though.

Reply:
This file defines several common project configurations, which can be applied to any subproject. 

Comparing to the solution we have for dataflow build, one benefit of using such configuration to create IT is that we can have one centralized place for all IO to set up ITs instead of having them one by one and it's easier to use one test configuration to invoke tests on different runners. Taking this KafkaIO IT as an example, if one day Flink is interested in this suite, Flink  can just use the same configuration in https://github.com/apache/beam/blob/master/.test-infra/jenkins/job_PerformanceTests_KafkaIO_IT.groovy.